src/lib.rs

```diff
@@ -37,11 +37,26 @@ mod errors {
                 description("unrecognizable charset")
                 display("uchardet was unable to recognize a charset")
             }
-            DataHandlingError {
-                description("data handling error")
-                display("uchardet failed to handle the supplied data (out of memory?)")
+            OutOfMemory {
+                description("out of memory error")
+                display("uchardet ran out of memory")
+            }
+            Other(int: i32) {
+                description("unknown error")
+                display("uchardet returned unknown error {}", int)
+            }
+        }
+    }
+
+    impl ErrorKind {
+        pub fn from_nsresult(nsresult: ::ffi::nsresult) -> ErrorKind {
```

Comment:
I actually also thought about that possibility but after taking a glance at [quick_error.rs](https://github.com/brson/error-chain/blob/master/src/quick_error.rs) I quickly abandoned that idea... :sweat_smile: 

(PR is up: https://github.com/brson/error-chain/pull/50)